modules/gateway/nodes.go

```diff
@@ -16,18 +16,29 @@ var (
 	errOurAddress = errors.New("can't add our own address")
 )
 
+type node struct {
+	Inbound bool `json:"inbound"`
+}
+
 // addNode adds an address to the set of nodes on the network.
-func (g *Gateway) addNode(addr modules.NetAddress) error {
+func (g *Gateway) addNode(addr modules.NetAddress, inbound bool) error {
 	if addr == g.myAddr {
 		return errOurAddress
-	} else if _, exists := g.nodes[addr]; exists {
+	} else if node, exists := g.nodes[addr]; exists {
+		if node.Inbound != inbound {
+			node.Inbound = inbound
+			return nil
+		}
```

Comment:
This whole bit of code I think could be rewritten as:

`node.Inbound = inbound; return nil`

But I'm not sure that's the desired behavior. Once we've marked a node as an outbound node, we should probably keep them marked as `outbound` permanently. Additionally, upon startup we should start by connecting to our existing outbound peers instead of just connecting around randomly. If we already have a set of trusted outbound peers, we should stick to that set instead of connecting to potential attackers.